experimentation/export_model_predictions.py

```diff
@@ -0,0 +1,148 @@
+from collections.abc import Iterable
+from dataclasses import dataclass
+from itertools import islice
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+from catboost import CatBoostClassifier, Pool
+from nxontology import NXOntology
+from sklearn.pipeline import Pipeline, make_pipeline
+
+from experimentation.model_utils import BiasedMaeMetric
+from nxontology_ml.data import get_efo_otar_slim, read_training_data
+from nxontology_ml.efo import NodeXrefFeatures
+from nxontology_ml.features import NodeInfoFeatures, PrepareNodeFeatures
+from nxontology_ml.model import CatBoostDataFormatter
+from nxontology_ml.text_embeddings.text_embeddings_transformer import (
+    TextEmbeddingsTransformer,
+)
+from nxontology_ml.utils import ROOT_DIR
+
+NON_DISEASE_THERAPEUTIC_AREAS: set[str] = {
+    "EFO:0005932",  # animal_disease
+    "EFO:0001444",  # measurement
+    "EFO:0000651",  # phenotype
+    "GO:0008150",  # biological_process
+    "EFO:0002571",  # medical_procedure
+}
+
+
+def get_disease_nodes(
+    take: int | None = None,
+    nxo: NXOntology[str] | None = None,
+) -> Iterable[str]:
+    """
+    If all roots are in the set of non-disease therapeutic areas (aka roots in EFO OTAR Slim),
+        then its non-disease
+    """
+    nxo = nxo or get_efo_otar_slim()
+    for n in islice(sorted(nxo.graph), take):
+        if any(
+            root not in NON_DISEASE_THERAPEUTIC_AREAS for root in nxo.node_info(n).roots
+        ):
+            yield n
+
+
+rs_cls_to_precision = {
+    "01-disease-subtype": "high",
+    "02-disease-root": "medium",
+    "03-disease-area": "low",
+}
+
+
+@dataclass
+class NodeLabelOutput:
+    """
+    Main output of the model inference tasks
+    """
+
+    identifier: str
+    precision: str
+    rs_classification: str | None
+    probas: list[float]
+
+
+def export_model_predictions(
+    precisions_file: Path = ROOT_DIR / "data/efo_otar_slim_v3.57.0_precisions.tsv",
+    features_file: Path = ROOT_DIR / "data/efo_otar_slim_v3.57.0_features.tsv",
```

Comment:
Why not combine into a single file?

Reply:
We should also include the node names/labels in both datasets as the second column (or single consolidated dataset)

Reply:
> Why not combine into a single file? Single file can contain training labels if those are readily available as well.

I find the feature file a little large/bloated and I'm not sure I understand how it's supposed to be used, unlike the labeled one. That's why I separated it. I can surely combine them.

Reply:
> We should also include the node names/labels in both datasets as the second column (or single consolidated dataset)

Okay